lib/types/result-set.js

```diff
@@ -140,8 +46,89 @@ function ResultSet(result) {
    * following pages of results.
    * @type {Array<Row>|undefined}
    */
-  // ToDo: Add undefined
   this.rows = getRowsFromResultsWrapper(result);
+
+  /**
+   * Gets the row length of the result, regardless if the result has been buffered or not
+   * @type {Number|undefined}
+   */
+  this.rowLength = this.rows.length;
+
+  /**
+   * Gets the columns returned in this ResultSet.
+   * @type {Array.<{name, type}>}
+   * @default null
+   */
+  this.columns = getColumnsMetadata(result);
+
+  /**
+   * A string token representing the current page state of query. It can be used in the following executions to
+   * continue paging and retrieve the remained of the result for the query.
+   * @type {String|null}
+   * @default null
+   */
+  this.pageState = null;
+
+  // Odl logic for that:
+  //     if (meta.pageState) {
+  //       this.pageState = meta.pageState.toString('hex');
+
+  //       // Expose rawPageState internally
+  //       Object.defineProperty(this, 'rawPageState', { value: meta.pageState, enumerable: false });
+  //     }
+
+  /**
+   * Information on the execution of a successful query:
+   * @member {Object}
+   * @property {Number} achievedConsistency The consistency level that has been actually achieved by the query.
+   * @property {String} queriedHost The Cassandra host that coordinated this query.
+   * @property {Object} triedHosts Gets the associative array of host that were queried before getting a valid response,
+   * being the last host the one that replied correctly.
+   * @property {Object} speculativeExecutions The number of speculative executions (not including the first) executed before
+   * getting a valid response.
+   * @property {Uuid} traceId Identifier of the trace session.
+   * @property {Array.<string>} warnings Warning messages generated by the server when executing the query.
+   * @property {Boolean} isSchemaInAgreement Whether the cluster had reached schema agreement after the execution of
+   * this query.
+   * <p>
+   * [[ToDo: Check if this is the case in rust driver]]
+   *   After a successful schema-altering query (ex: creating a table), the driver will check if
+   *   the cluster's nodes agree on the new schema version. If not, it will keep retrying for a given
+   *   delay (see <code>protocolOptions.maxSchemaAgreementWaitSeconds</code>).
+   * </p>
+   * <p>
+   *   Note that the schema agreement check is only performed for schema-altering queries For other
+   *   query types, this method will always return <code>true</code>. If this method returns <code>false</code>,
+   *   clients can call [Metadata.checkSchemaAgreement()]{@link module:metadata~Metadata#checkSchemaAgreement} later to
+   *   perform the check manually.
+   * </p>
+   */
+  this.info = {
+    queriedHost: undefined, // Is supported by rust driver?
+    triedHosts: undefined, // Is supported by rust driver?
+    speculativeExecutions: undefined, // Is supported by rust driver?
```

Comment:
These are not collected by default, but could be acquired by use of a custom [`HistoryListener`](https://docs.rs/scylla/latest/scylla/history/trait.HistoryListener.html).